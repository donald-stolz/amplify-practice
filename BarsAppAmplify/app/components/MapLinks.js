// @flow
import React from 'react';
import { Popup } from 'react-native-map-link';

// Types
type Props = {
  isVisible: boolean,
  onCancelPressed: Function,
  onAppPressed: Function,
  onBackButtonPressed: Function,
  name: string,
  lat: number,
  lng: number,
  barId: string,
};

const MapLinks = ({
  isVisible,
  onCancelPressed,
  onAppPressed,
  onBackButtonPressed,
  name,
  lat,
  lng,
  barId,
}: Props) => (
  <Popup
    isVisible={isVisible}
    onCancelPressed={onCancelPressed}
    onAppPressed={onAppPressed}
    onBackButtonPressed={onBackButtonPressed}
    modalProps={{
      // you can put all react-native-modal props inside.
      animationIn: 'slideInUp',
    }}
    appsWhiteList={[
      'apple-maps',
      'google-maps',
      'uber',
      'waze',
      'lyft',
      'transit',
      'yandex',
      'moovit',
      'citymapper',
    ]}
    options={{
      /* See `showLocation` method above, this accepts the same options. */
      title: name,
      latitude: lat,
      longitude: lng,
      googleForceLatLon: true,
      googlePlaceId: barId,
      dialogTitle: 'Directions',
      dialogMessage: 'Open bar location in one of the following apps:',
      cancelText: 'Cancel',
    }}
    style={
      {
        /* Optional: you can override default style by passing your values. */
      }
    }
  />
);

export default MapLinks;
