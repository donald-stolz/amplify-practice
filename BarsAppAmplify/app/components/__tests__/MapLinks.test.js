import React from 'react';
import renderer from 'react-test-renderer';
import MapLinks from '../MapLinks';

test('MapLinks renders correctly', () => {
  const onCancelPressed = jest.fn();
  const onAppPressed = jest.fn();
  const onBackButtonPressed = jest.fn();
  const tree = renderer
    .create(
      <MapLinks
        isVisible
        onCancelPressed={onCancelPressed}
        onAppPressed={onAppPressed}
        onBackButtonPressed={onBackButtonPressed}
        name="Credo"
        lat={123123123}
        lng={444444444}
        barId="123456789"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
