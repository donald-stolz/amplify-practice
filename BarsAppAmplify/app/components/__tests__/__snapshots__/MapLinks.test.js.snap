// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapLinks renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={true}
  onModalHide={[Function]}
  onRequestClose={[MockFunction]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#464646",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={true}
    onModalHide={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "maxHeight": 800.4,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#ECF2F8",
              "borderColor": "transparent",
              "borderWidth": 1,
              "padding": 15,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#464646",
                "fontSize": 16,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Directions
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#ACBBCB",
                "fontSize": 12,
                "marginTop": 10,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Open bar location in one of the following apps:
        </Text>
      </View>
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "transparent",
            "borderTopColor": "#ECF2F8",
            "borderWidth": 1,
            "justifyContent": "center",
            "opacity": 1,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#BBC4CC",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
